flocker/control/_config.py

```diff
@@ -685,11 +685,19 @@ def _link_applications(self):
                 for target_ports_object in target_application_ports:
                     local_port = target_ports_object.internal_port
                     remote_port = target_ports_object.external_port
-                    app_links.append(
-                        Link(local_port=local_port,
-                             remote_port=remote_port,
-                             alias=link_definition['alias'])
-                    )
+                    try:
+                        app_links.append(
+                            Link(local_port=local_port,
+                                 remote_port=remote_port,
+                                 alias=link_definition['alias'])
+                        )
+                    except InvariantException:
+                        raise ConfigurationError(
+                            ("Application '{application}' has a config "
+                             "error. 'links' alias names must be "
+                             "alphanumeric.").format(
+                                 application=application_name)
```

Comment:
Given difficulty of extracting useful information from `InvariantException`, maybe just logging the exception that got raised is sufficient (... er except do logs from this code go anywhere? :/)